osafw-app/App_Code/fw/ConvUtils.cs

```diff
@@ -236,4 +241,199 @@ private static string _replace_specials(string html_data)
+        var columns = new DocumentFormat.OpenXml.Spreadsheet.Columns();
+        double maxWidth = 10;
+
+        UInt32Value iter = 1;
+        foreach (string item in headers)
+        {
+            var val = maxColWidth[item];
+            var width = ((val * maxWidth + 5) / maxWidth * 256) / 256;
+            DocumentFormat.OpenXml.Spreadsheet.Column col = new DocumentFormat.OpenXml.Spreadsheet.Column();
+            col.BestFit = true;
+            col.Min = iter;
+            col.Max = iter;
+            col.CustomWidth = true;
+            col.Width = (double)width;
+            columns.Append(col);
+            iter += 1;
+        }
+        return columns;
+    }
+
+    public static Stylesheet GetStylesheet()
+    {
+        var _Fonts = new DocumentFormat.OpenXml.Spreadsheet.Fonts();
+        _Fonts.Append(new DocumentFormat.OpenXml.Spreadsheet.Font());
+        _Fonts.Append(new DocumentFormat.OpenXml.Spreadsheet.Font(new DocumentFormat.OpenXml.Spreadsheet.Bold()));
+        _Fonts.Append(new DocumentFormat.OpenXml.Spreadsheet.Font(new DocumentFormat.OpenXml.Spreadsheet.Bold()));
+
+        var _Fills = new DocumentFormat.OpenXml.Spreadsheet.Fills(new DocumentFormat.OpenXml.Spreadsheet.Fill());
+        var _Borders = new DocumentFormat.OpenXml.Spreadsheet.Borders(new DocumentFormat.OpenXml.Spreadsheet.Border());
+        _Borders.Append(new DocumentFormat.OpenXml.Spreadsheet.Border());
+
+        var border = new DocumentFormat.OpenXml.Spreadsheet.Border();
+        var topBorder = new DocumentFormat.OpenXml.Spreadsheet.TopBorder();
+        topBorder.Style = BorderStyleValues.Medium;
+        border.TopBorder = topBorder;
+        _Borders.Append(border);
+
+        var _CellFormats = new DocumentFormat.OpenXml.Spreadsheet.CellFormats();
+        _CellFormats.Append(new DocumentFormat.OpenXml.Spreadsheet.CellFormat());
+
+        var cellFormat = new DocumentFormat.OpenXml.Spreadsheet.CellFormat();
+        cellFormat.FontId = 1;
+        cellFormat.FillId = 0;
+        cellFormat.BorderId = 1;
+        _CellFormats.Append(cellFormat); // header
+
+        cellFormat = new DocumentFormat.OpenXml.Spreadsheet.CellFormat();
+        cellFormat.FontId = 1;
+        cellFormat.FillId = 0;
+        cellFormat.BorderId = 2;
+        _CellFormats.Append(cellFormat); // footer
+
+        return new Stylesheet(_Fonts, _Fills, _Borders, _CellFormats);
+    }
+
+
+    /// <summary>
+    /// Create xls file form hashtable where key is sheet name and value is rows array of hashtables with table data
+    /// * <param name="xls_export_headers">XLS headers row, comma-separated format</param>
+    /// * <param name = "xls_export_fields" > empty, * or Utils.qw format</param>
+    /// * <param name = "rows" > DB array</param>
+    /// </summary>
+    /// <param name="sheetsData"></param>
+    public static string parsePageNativeExcel(FW fw, string xls_export_headers, string xls_export_fields, ArrayList rows, string out_filename = "")
```

Comment:
make output similar to other parsePage methods - depending on out_filename.
I.e. if it's empty - output to browser with proper headers